frontend/src/app/[lng]/member/edit/ValidateEmail.tsx

```diff
@@ -0,0 +1,123 @@
+"use client";
+
+import { useImperativeHandle, useRef } from "react";
+import { Button as ThemeButton, Grid } from "@radix-ui/themes";
+import {
+  DialogRoot,
+  DialogContent,
+  DialogHeader,
+  DialogBody,
+  DialogFooter,
+} from "@/app/components/Dialog";
+import Button from "@/app/components/Button";
+import TextField from "@/app/components/TextField";
+import OTPDialog, { OTPImperativeHandle } from "@/app/[lng]/login-process/otp";
+import useDialogWithForm, {
+  CLOSE_REJECT,
+} from "@/app/[lng]/login-process/useDialogWithForm";
+import { useT } from "@/app/i18n/client";
+
+export default function ValidateEmail() {
+  const { t: memberTranslate } = useT("member");
+  const otpDialogRef = useRef<OTPImperativeHandle>(null);
+  const newEmailOtpDialogRef = useRef<OTPImperativeHandle>(null);
+  const newEmailDialogRef = useRef<OTPImperativeHandle>(null);
+
+  const handleEditEmail = async () => {
+    try {
+      await otpDialogRef.current?.openDialogWithPromise();
+      await newEmailDialogRef.current?.openDialogWithPromise();
+      await newEmailOtpDialogRef.current?.openDialogWithPromise();
+    } catch (error) {
+      if (error === CLOSE_REJECT) {
+        return;
+      }
+      if (!error || !(error instanceof Error)) {
+        console.error("Unexpected error:", error);
+        return;
+      }
+    }
+  };
+
+  return (
+    <>
+      <ThemeButton
+        size="2"
+        my="1"
+        ml="2"
+        variant="surface"
+        className="self-end cursor-pointer"
+        onClick={handleEditEmail}
+      >
+        {memberTranslate("edit.email-edit")}
+      </ThemeButton>
+      <OTPDialog
+        ref={otpDialogRef}
+        title={memberTranslate("edit.original-email-otp")}
+        buttonText={memberTranslate("edit.button-validate")}
+      />
+      <OTPDialog
+        ref={newEmailOtpDialogRef}
+        title={memberTranslate("edit.new-email-otp")}
+        buttonText={memberTranslate("edit.button-validate")}
+      />
+      <NewEmailDialog ref={newEmailDialogRef} />
+    </>
+  );
+}
+
+function NewEmailDialog({
+  ref,
```

Comment:
好奇查了一下，如果要接收直接接收父層傳進來的 ref 的話，AI 是說需要使用 `forwardRef`允許函式型元件接收 ref，並轉給內部，可以給你參考看看～
```suggestion
 function NewEmailDialog = forwardRef<OTPImperativeHandle, {}>((props, ref) => {...}
```

Reply:
React v19 不需要 `forwardRef` 了！（可喜可賀）
附上[官方的版本更新說明](https://react.dev/blog/2024/12/05/react-19#improvements-in-react-19)